Improving database logging

locnh opened the pull request:
Hi everyone,

I'm using PHPIDS for security monitoring and collect attacking events for analysing system. So, I found that PHPIDS need to log more and more infomation about attacker's behaviors.

I wrote a few lines of code to improve database logger, then PHPIDS can log POST payload, it should be convenient for auditing and learning on attacker's behaviors.

Thanks and regards.

locnh committed:
Improve database logging

Database logger can log POST payload, it should be convenient for
auditing and learning on attacker's behaviors.

lstrojny commented:
Thanks for your contribution. Could you add an integration test that uses PDO and SQLite as a backend?

locnh commented:
Yes, sure. I will test and commit soon ! :smiley:

locnh committed:
SQLite supported

- SQLite supported
- Request params, User-agent, Cookie, Referer logged

locnh commented:
Hi Lars Strojny,
PHPIDS is now SQLite supported.
All tested on my productions :+1:

lstrojny commented on lib/IDS/Log/Database.php:
```diff
@@ -36,22 +36,53 @@
+  `origin` varchar(15) NOT NULL,
+  `created` datetime NOT NULL,
+  PRIMARY KEY (`id`)
+) ENGINE=MyISAM DEFAULT CHARSET=latin1 AUTO_INCREMENT=1 ;
```
We should use InnoDB here

locnh replied:
I missed, in case we use PHPIDS as a censor on many servers but using a central DB, MyISAM will be stuck on table lock, if we are in big wave of scanning or attacking.

Thanks Lars.

locnh committed:
Should use InnoDB for DB logging

In case we use PHPIDS as a censor on many servers but using a central
DB, MyISAM will be stuck on table lock, if we are in big wave of
scanning or attacking.

lstrojny commented:
What about the unit tests?

locnh commented:
Unit tests ? I don't understand what you mean.

locnh committed:
Update docs/sql/intrusions.sql

locnh committed:
Update docs/sql/intrusions.sql

lstrojny commented:
The logging part of PHPIDS will be removed in the future. I don't accept any new pull requests for it. Sorry.
